crates/gen-guest-c/src/lib.rs

```diff
@@ -221,12 +221,7 @@ impl WorldGenerator for C {
         if self.needs_string {
             uwrite!(
                 h_str,
-                "
-                   typedef struct {{
-                       {ty} *ptr;
-                       size_t len;
-                   }} {snake}_string_t;
-               ",
+                "typedef struct {{\n  {ty} *ptr;\n  size_t len;\n}} {snake}_string_t;\n",
```

Comment:
How come this changed? I personally find it easier to read the prior version and would prefer to keep that.

If all that's necessary is removing the leading newline then you can define the string as `"\` at the start which trims the leading newline.

(same for the adjustments below, I'd prefer to keep the multiline version if possible)

Reply:
This is actually quite a specific output bug where the internal formatter needs a workaround. I've added a comment to give some more context. I think because of the `extern C {` brace being flattened, the source indentation system is getting confused. It was outputting:

```c
#ifndef __BINDINGS_LEXER_WORLD_H
#define __BINDINGS_LEXER_WORLD_H
#ifdef __cplusplus
extern "C" {
#endif

#include <stdlib.h>
#include "lexer_world.h"
#include <string.h>
#include <uchar.h>
#include <stdint.h>
#include <stdbool.h>

                   typedef struct {
                     char16_t *ptr;
                     size_t len;
                   } lexer_world_string_t;

typedef struct {
  bool is_some;
  lexer_world_string_t val;
} lexer_world_option_string_t;
```

It was only affecting that first declaration, and all subsequent indentation was correct in being at the base level (instead of the entire source being indented by the `extern C {` which I fixed in a previous PR.

With this change this fixes the output to properly indent:

```c
#ifndef __BINDINGS_LEXER_WORLD_H
#define __BINDINGS_LEXER_WORLD_H
#ifdef __cplusplus
extern "C" {
#endif

#include <stdlib.h>
#include "lexer_world.h"
#include <string.h>
#include <uchar.h>
#include <stdint.h>
#include <stdbool.h>

typedef struct {
  char16_t *ptr;
  size_t len;
} lexer_world_string_t;

typedef struct {
  bool is_some;
  lexer_world_string_t val;
} lexer_world_option_string_t;
```

Regressions could be avoided on this work by having proper code generation output snapshotting.

Reply:
The other corrections were actually a similar issue.

Reply:
Could this be fixed in `Source` rather than making the source less readable?

Reply:
Okay, I was able to resolve this by using the `source.deindent()` function.